### PR TITLE
Python: Fix attribute taint

### PR DESCRIPTION
**Pull request by RasmusWL**
Fixes https://github.com/github/codeql/issues/7786

Draft PR so we can gauge what the impact of this change is before merging it (and look for failing QL tests :sweat_smile:)

### CODE REVIEW

**Comment by RasmusWL** [comment]
Judging from the results from the internal experiment, I think this looks good :+1:

File: python/ql/test/experimental/dataflow/tainttracking/defaultAdditionalTaintStep/test_with.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def __exit__(self, exc_type, exc, tb):
 def test_with_arg():
     ctx = Context_arg(TAINTED_STRING)
     with ctx as tainted:
-        ensure_tainted(tainted) # $ tainted
+        ensure_tainted(tainted) # $ MISSING: tainted
```
Comment (yoff) [inline python/ql/test/experimental/dataflow/tainttracking/defaultAdditionalTaintStep/test_with.py:52]: This is actually expected, I think, until we model `__enter__`.

**Review by yoff — COMMENTED** [review]
Basically looks fine. Is [this removed alert](https://github.com/github/codeql-dca-main/blob/data/RasmusWL/pr-7873-b276b2d__nightly__code-scanning/reports/alert-comparison.md#--pyclear-text-logging-sensitive-data-for-modularhistory__modularhistory-3) an FP in this context?

File: python/ql/test/experimental/dataflow/tainttracking/defaultAdditionalTaintStep/test_with.py (hunk 1 above)
Comment (RasmusWL) [reply python/ql/test/experimental/dataflow/tainttracking/defaultAdditionalTaintStep/test_with.py:52]: agreed :+1:

**Comment by RasmusWL** [comment]
> Basically looks fine. Is [this removed alert](https://github.com/github/codeql-dca-main/blob/data/RasmusWL/pr-7873-b276b2d__nightly__code-scanning/reports/alert-comparison.md#--pyclear-text-logging-sensitive-data-for-modularhistory__modularhistory-3) an FP in this context?

yes, that's a FP. The current path is:

1: ControlFlowNode for cert_file

image

2: [post store] ControlFlowNode for self

image

3: ControlFlowNode for Attribute() 

image

4: ControlFlowNode for Fstring 

image

This PR means that we no longer treat the `connection` as tainted with a certificate, which seems like a good improvement :+1:

**Comment by RasmusWL** [comment]
> Basically looks fine. Is [this removed alert](https://github.com/github/codeql-dca-main/blob/data/RasmusWL/pr-7873-b276b2d__nightly__code-scanning/reports/alert-comparison.md#--pyclear-text-logging-sensitive-data-for-modularhistory__modularhistory-3) an FP in this context?

yes, that's a FP. The current path is:

1: ControlFlowNode for cert_file

image

2: [post store] ControlFlowNode for self

image

3: ControlFlowNode for Attribute() 

image

4: ControlFlowNode for Fstring 

image

This PR means that we no longer treat the `connection` as tainted with a certificate, which seems like a good improvement :+1:

**Comment by yoff** [comment]
Cool, thanks for showing the path.

**Review by yoff — APPROVED** [review]
LGTM, always nice to see fixes like these.

**Pull request merged**